messages/eslintrc-plugins.js

```diff
@@ -0,0 +1,22 @@
+"use strict";
+
+module.exports = function({ plugins }) {
+
+    return `
+A config object has a "plugins" key defined as an array of strings.
+
+Flat config requires "plugins" to be an object in this form:
+
+    {
+        plugins: {
+            ${plugins[0] || "namespace"}: pluginObject
+        }
+    }
```

Comment:
Flat config schema throws this error on any array (it doesn't check whether it's an array of _strings_), so if the user specifies plugins like this:

```js
module.exports = [{
    plugins: [somePluginObject]
}];
``` 

The printed message will look like this:

```
Oops! Something went wrong! :(

ESLint: 8.45.0


A config object has a "plugins" key defined as an array of strings.

Flat config requires "plugins" to be an object in this form:

    {
        plugins: {
            [object Object]: pluginObject
        }
    }

Please see the following page for information on how to convert your config object into the correct format:
https://eslint.org/docs/latest/use/configure/migration-guide#importing-plugins-and-custom-parsers

If you're using a shareable config that you cannot rewrite in flat config format, then use the compatibility utility:
https://eslint.org/docs/latest/use/configure/migration-guide#using-eslintrc-configs-in-flat-config
```

Perhaps we should have two templates?

* `plugins` is an array of strings (or empty array?) - possible eslintrc config, use this template
* `plugins` is an array but not array of strings - use another template.

Reply:
I think for now I'd like to keep things simple and just assume that when people are using an array then they are most likely using an eslintrc-style config. We can always expand later if we find that's not the case.

I've updated to check if the first item in the array is a string to make sure that the message makes sense.